libraries/AP_Arming/AP_Arming.cpp

```diff
@@ -168,10 +168,25 @@ bool AP_Arming::barometer_checks(bool report)
 {
     if ((checks_to_perform & ARMING_CHECK_ALL) ||
         (checks_to_perform & ARMING_CHECK_BARO)) {
-        if (!AP::baro().all_healthy()) {
+        const AP_Baro *baro = AP_Baro::get_singleton();
+        if (!baro->all_healthy()) {
             check_failed(ARMING_CHECK_BARO, report, "Barometer not healthy");
             return false;
         }
+        if (checks_to_perform & ARMING_CHECK_GPS) {
+            const AP_GPS *gps = AP_GPS::get_singleton();
```

Comment:
If you call get_singleton() you need to check it for null pointer. If you use the `AP::gps()` method to get a reference then you don't need a null pointer check.

Reply:
Done Thanks.